connectors/radon.py

```diff
@@ -0,0 +1,251 @@
+        self.loc = []
+        self.lloc = []                          
+        self.sloc = []                          
+        self.comments = []                      
+        self.docstring = []                     
+        self.blank = []                         
+        self.single_comment = []                     
+        self.noc = []                           
+        self.nom = []                           
+        self.nof = []                           
+        self.class_loc = []                     
+        self.method_loc = []                    
+        self.func_loc = []        
+        # calcule
+        self.calcul_noc = 0
+        self.calcul_nom = 0
+        self.calcul_nof = 0              
+
+    def analyze_source_code(self) -> None:
+        """
+        Analyzes the Python source code using the Radon library.
+
+        Args:
+        - None.
+
+        Returns:
+        - None.
+        """
+        
+        logging.info('RADON::analyze_repo')
+        metric = self.session.query(Metric).filter(Metric.version_id == self.version.version_id).first()
+        if (not metric):
+            metric = Metric()
+        
+        # TODO it will maybe need to be updated
+        logging.info('RADON self.config.language = ' + self.config.language)
+        if (self.config.language.lower() != "python"):
+            logging.info('RADON is only used for Python language')
+        else:
+            if (not metric.radon_cc_total):
+                self.compute_metrics()
+                self.__store_metrics(metric)
+            else:
+                logging.info('RADON analysis already done for this version, version: ' + str(self.version.version_id))
+
+    def compute_metrics(self) -> None:
+        """
+        Computes the software metrics for the Python codebase.
+
+        Args:
+        - None
+
+        Returns:
+        - None.
+        """
+        for root, dirs, files in os.walk(self.directory):
+            for file in files:
+                file_path = os.path.join(root, file)
+                if (is_python_file(file)):
+                    with open(file_path, "r", encoding="utf-8") as file:
+                        code = file.read()
+
+                    # Recovery of radon metrics
+                    cc_metrics = cc_visit(code)
+                    raw_metrics = analyze(code)
+
+                    # Processing of recovered metrics
+                    self.avg_cc.append(average_complexity(cc_metrics))
+                    self.__compute_cc_metrics(cc_metrics)
+                    self.__compute_raw_metrics(raw_metrics)
+        logging.info('Adding Radon analysis for this version, version: ' + str(self.version.version_id))
+
+
+    def __compute_cc_metrics(self, cc_metrics) -> None:
+
+            self.__calcule_cc_metrics(cc_metrics)
+            self.noc.append(self.calcul_noc)
+            self.nom.append(self.calcul_nom)
+            self.nof.append(self.calcul_nof)
```

Comment:
Misuse of object attributes in my opinion. The `calcul_*` should be local variables and be returned by __calcule_cc_metrics.